model/config.go

```diff
@@ -504,9 +504,14 @@ type JobSettings struct {
 	RunScheduler *bool
 }
 
+type PluginState struct {
+	Enable bool
+}
+
 type PluginSettings struct {
-	Enable  *bool
-	Plugins map[string]interface{}
+	Enable       *bool
+	Plugins      map[string]interface{}
+	PluginStates map[string]*PluginState
```

Comment:
Not entirely happy with this, open to suggestions for doing it differently

Reply:
Assuming we don't care about backwards compatibility, maybe something a bit more hierarchical like...

```go
type InstalledPluginSettings struct {
    State *PluginState
    Settings interface{}
}

type PluginSettings struct {
    Enable *bool
    Plugins map[string]*InstalledPluginSettings
}
```

/shrug